main.go

```diff
@@ -211,15 +211,17 @@ func WaitDaemonsetReady(namespace, name string, timeout time.Duration) error {
 			return fmt.Errorf("failed to get daemonset, err: %s", err)
 		}
 
-		if daemonSet.Status.DesiredNumberScheduled != nodesCount {
-			return fmt.Errorf("daemonset DesiredNumberScheduled not equal to number of nodes:%d, please instantiate debug pods on all nodes", nodesCount)
+		if daemonSet.Status.DesiredNumberScheduled == nodesCount {
+			logrus.Infof("Waiting for (%d) debug pods to be ready: %+v", nodesCount, daemonSet.Status)
+			if isDaemonSetReady(&daemonSet.Status) {
+				isReady = true
+				break
+			}
+		} else {
+			logrus.Warnf("daemonset DesiredNumberScheduled %d not equal to number of nodes:%d, please instantiate debug pods on all nodes",
```

Comment:
Same as before! "debug pods" is misleading.

Reply:
Done. Also I changed the "please instantiate..." to "please make sure daemonset pods can be deployed on al nodes."

Reply:
Now I am confused. How can one make sure that Daemonset pods are deployed on all nodes?

Reply:
"... that pods **can be** deployed on all nodes": make sure there's nothing in your cluster (SAs, RBACs or whatever weird config) preventing this privilege daemonset to deploy its pods on all the nodes.

Reply:
Why don't we change the statement to something simple? ```logrus.Warnf("daemonset %s could not be deployed.", daemonSetName)```

Reply:
I agree, the values are very helpful.

Reply:
Liked the newest message. Now, I got it - the issue refers to 0 set to desiredNumberScheduled.

Reply:
Done.